### DISCUSSION TITLE
useFieldArray changes defaultValues of useForm

### DISCUSSION THREAD

**Discussion by ryanmk54** [react-hook-form/react-hook-form#6919] [Q&A] [answered]
I have a problem where changing the value of an input in useFieldArray changes the defaultValues object passed into useForm. The main problem is that isDirty is still false, because the input value equals the value in defaultValues. The code below is a simple example of where it is happening:
```
  const buildTextField = ({label, name, inputProps = {}, required = false}) => (
    <TextField
      error={Boolean(errors?.[name])}
      fullWidth
      label={label}
      disabled={props.disabled}
      inputProps={{...inputProps, ...register(name, {required})}}
   />
  );
```
I tried to create a reproducible code sandbox [here](https://codesandbox.io/s/nifty-tereshkova-bny5k?file=/src/App.js), but it works correctly in the codesandbox. I am using react-query to create the object used for defaultValues, but the problem only happens in the useFieldArray fields. The fieldArray is created using the code below:
```
  const fieldArray = useFieldArray({
    control: props.form.control,
    keyName: "fieldId",
    name: "details"
  });
```
The form is created using `const form = useForm({defaultValues})`

Any guesses as to what I might be doing wrong, or what else I could try?

**Comment by ryanmk54** [comment]
I was able to find that the dirty state in useFieldArray fields
is set properly in 7.15 - 7.17.1
is broken 7.17.2 - 7.18

It works in the codesandbox. I'm not sure what's different in my app.

**Reply by bluebill1049** [reply]
thanks for the investiation. can you reproduce the problem in codesandbox?

**Reply by bluebill1049** [reply]
i can probably then looking into the release: https://github.com/react-hook-form/react-hook-form/releases/tag/v7.17.2 if it's introduced in 7.17.2

**Comment by ryanmk54** [comment, marked as answer]
To add an answer to this discussion, this is the same as
* https://github.com/react-hook-form/react-hook-form/issues/7723.
* https://github.com/react-hook-form/react-hook-form/discussions/7710#discussioncomment-2098582
